_health-care/_js/components/personal-information/State.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import _ from 'lodash';
+
+/**
+ * Select component for the US state of an address.
+ *
+ * No validation is provided since base UI elements provide all
+ * common options as provided in reference PDF/JS form.
+ */
+class State extends React.Component {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentWillMount() {
+    this.selectId = _.uniqueId('state-select-');
+  }
+
+  handleChange() {
+    this.props.onUserInput(this.refs.state.value);
+  }
+
+  render() {
+    const arrayOfState = {
+      AL: 'Alabama',
+      AK: 'Alaska',
+      AS: 'American Samoa',
+      AZ: 'Arizona',
+      AR: 'Arkansas',
+      CA: 'California',
+      CO: 'Colorado',
+      CT: 'Connecticut',
+      DE: 'Delaware',
+      DC: 'District Of Columbia',
+      FM: 'Federated States Of Micronesia',
+      FL: 'Florida',
+      GA: 'Georgia',
+      GU: 'Guam',
+      HI: 'Hawaii',
+      ID: 'Idaho',
+      IL: 'Illinois',
+      IN: 'Indiana',
+      IA: 'Iowa',
+      KS: 'Kansas',
+      KY: 'Kentucky',
+      LA: 'Louisiana',
+      ME: 'Maine',
+      MH: 'Marshall Islands',
+      MD: 'Maryland',
+      MA: 'Massachusetts',
+      MI: 'Michigan',
+      MN: 'Minnesota',
+      MS: 'Mississippi',
+      MO: 'Missouri',
+      MT: 'Montana',
+      NE: 'Nebraska',
+      NV: 'Nevada',
+      NH: 'New Hampshire',
+      NJ: 'New Jersey',
+      NM: 'New Mexico',
+      NY: 'New York',
+      NC: 'North Carolina',
+      ND: 'North Dakota',
+      MP: 'Northern Mariana Islands',
+      OH: 'Ohio',
+      OK: 'Oklahoma',
+      OR: 'Oregon',
+      PW: 'Palau',
+      PA: 'Pennsylvania',
+      PR: 'Puerto Rico',
+      RI: 'Rhode Island',
+      SC: 'South Carolina',
+      SD: 'South Dakota',
+      TN: 'Tennessee',
+      TX: 'Texas',
+      UT: 'Utah',
+      VT: 'Vermont',
+      VI: 'Virgin Islands',
+      VA: 'Virginia',
+      WA: 'Washington',
+      WV: 'West Virginia',
+      WI: 'Wisconsin',
+      WY: 'Wyoming'
+    };
+    const options = _.map(arrayOfState, (val, key) => {
+      return (
+        <option
+            key={key}
```

Comment:
`key=` is required by React when constructing `<options>` from a map this way. I got nasty warnings when I left this out and scratched my head looking at the URL in the warning. Basically, the other way of generating a `<select>` (like in the `Gender` component), there are magically generated string/IDs for each option/value. `key=` in this context give React a hint at the mapping between my values and its own.